Makefile

```diff
@@ -662,61 +794,46 @@ install.libseccomp.sudo:
 	git clone https://github.com/seccomp/libseccomp ../../seccomp/libseccomp
 	cd ../../seccomp/libseccomp && git checkout --detach $(LIBSECCOMP_COMMIT) && ./autogen.sh && ./configure --prefix=/usr && make all && make install
 
-
-.PHONY: completions
-completions: podman podman-remote
-	# key = shell, value = completion filename
-	declare -A outfiles=([bash]=%s [zsh]=_%s [fish]=%s.fish [powershell]=%s.ps1);\
-	for shell in $${!outfiles[*]}; do \
-	    for remote in "" "-remote"; do \
-	        podman="podman$$remote"; \
-	        outfile=$$(printf "completions/$$shell/$${outfiles[$$shell]}" $$podman); \
-	        ./bin/$$podman completion $$shell >| $$outfile; \
-	    done;\
+.PHONY: uninstall
+uninstall:
+	for i in $(filter %.1,$(MANPAGES_DEST)); do \
+		rm -f $(DESTDIR)$(MANDIR)/man1/$$(basename $${i}); \
+	done; \
+	for i in $(filter %.5,$(MANPAGES_DEST)); do \
+		rm -f $(DESTDIR)$(MANDIR)/man5/$$(basename $${i}); \
 	done
+	# Remove podman and remote bin
+	rm -f $(DESTDIR)$(BINDIR)/podman
+	rm -f $(DESTDIR)$(BINDIR)/podman-remote
+	# Remove related config files
+	rm -f ${DESTDIR}${ETCDIR}/cni/net.d/87-podman-bridge.conflist
+	rm -f ${DESTDIR}${TMPFILESDIR}/podman.conf
+	rm -f ${DESTDIR}${SYSTEMDDIR}/io.podman.socket
+	rm -f ${DESTDIR}${USERSYSTEMDDIR}/io.podman.socket
+	rm -f ${DESTDIR}${SYSTEMDDIR}/io.podman.service
+	rm -f ${DESTDIR}${SYSTEMDDIR}/podman.service
+	rm -f ${DESTDIR}${SYSTEMDDIR}/podman.socket
+	rm -f ${DESTDIR}${USERSYSTEMDDIR}/podman.socket
+	rm -f ${DESTDIR}${USERSYSTEMDDIR}/podman.service
```

Comment:
Should -f be used? If uninstall is run shouldn't it report if some file cannot be removed?

Reply:
`-f` is always a good idea, especially so if there's no `-r`. In this case, the goal is: "I want `make uninstall` to pass cleanly, even if I haven't run `make install`, or even if I ran `make install` with an older Makefile that didn't install some of the files I'm now removing. I do not want `rm` to fail if /foo/bar/podman.socket does not exist". `-f` will not hide errors:
```console
rootless$ /bin/rm -f /etc/passwd;echo $?
/bin/rm: cannot remove '/etc/passwd': Permission denied
1
```
The only thing `-f` does is "remove even if file is not writable, and shut up if file does not exist"

Reply:
@edsantiago True `-f` does ignore missing files it also ignores all other errors. If the rm failed on an existing file, you have to intervene manually, which is why I called it out. Not that it's use is wrong, but can lead to unexpected results.

https://github.com/coreutils/coreutils/blob/master/src/rm.c#L334

Reply:
Yeah, I tend to agree with Ed.  I'm also not certain these have the same meaning for all contexts where the Makefile is used:

```
some_target:
    -rm -r /some /list /of /things

other_target:
    rm -rf /some /list /of /things
```

Reply:
Note: I could see a case for doing `-rm -rf /some /list /of /things`, especially for "clean".  But I'm lazy and don't want to wait for all the tests to run again :grin: